### PR TITLE
Gtw/clean_up_tooling

### PR DESCRIPTION
**Pull request by gregertw**
* Factor away .example stuff from data-model-examples
* Clean up deploy.sh

### CODE REVIEW

File: deploy.py [hunk 1]
```diff
@@ -43,18 +41,34 @@ def run(build_dir: str) -> None:
...
+    if (models_dir := Path(f"{build_dir}/domain_models")).is_dir():
+        load_datamodel_dump(
+            ToolGlobals, drop=drop, directory=models_dir, dry_run=dry_run
         )
-    if (models_dir := Path(f"{build_dir}/data_models")).is_dir():
+    if (models_dir := Path(f"{build_dir}/solution_models")).is_dir():
         load_datamodel_dump(
-            ToolGlobals, drop=True, directory=models_dir, dry_run=True
+            ToolGlobals, drop=drop, directory=models_dir, dry_run=dry_run
         )
```
Comment (doctrino) [inline deploy.py:72]: Reintroducing source, domain, and solution models at the top level of a build package?

File: modules/cdf_apm_simple/config.yaml [hunk 2]
```diff
@@ -1,4 +1,13 @@
 # Only valid for this module, loads template variables from environment
+raw_db: tutorial_apm
+datamodel: tutorial_apm_simple
+space: tutorial_apm_simple
+data_set: Valhall_System_23
+data_set_desc: Valhall_System_23
+datamodel_version: "1"
+view_Asset_version: "1"
+view_WorkOrder_version: "1"
+view_WorkItem_version: "1"
```
Comment (doctrino) [inline modules/cdf_apm_simple/config.yaml:10]: Great

File: scripts/datamodel.py [hunk 3]
```diff
@@ -237,10 +253,18 @@ def load_datamodel_dump(
     for type_, resources in cognite_resources_by_type.items():
         print(f"  {type_}: {len(resources)}")
 
-    space_list = list({r.space for _, resources in cognite_resources_by_type.items() for r in resources})
+    space_list = list(
+        {
+            r.space
+            for _, resources in cognite_resources_by_type.items()
+            for r in resources
+        }
```
Comment (doctrino) [inline scripts/datamodel.py:261]:
```suggestion
        {
            r.space
            for resources in cognite_resources_by_type.values()
            for r in resources
        }
```

**Review by doctrino — APPROVED** [review]
Seems like we have a fighting linter, as there are a few style changes. Will add some pre-commit checks when I get back to avoid this.

File: deploy.py (hunk 1 above)
Comment (gregertw) [reply deploy.py:72]: You hadn't merged them into top level, so I just temporary fixed the loading.

Comment (gregertw) [reply deploy.py:72]: That wasn't very clearly explained: You HAD merged into top-level, but when doing a build, each of the sub-folders with data models is merged into one for each data model type. We need to implement support for ordering import of modules, but temporary, I just implemented step-wise loading of modules in the deploy.sh script.

File: scripts/datamodel.py (hunk 3 above)
Comment (gregertw) [reply scripts/datamodel.py:261]: Sorry, auto-merge was on, I will turn that off, it's not particularly good practice unless the change is tiny.

**Pull request merged**